File: CSharp/Enterprise/ObjectRelational/Repository/Customer.cs
using System;

namespace Enterprise.ObjectRelational.Repository;

/// <summary>
/// Example domain entity.
/// </summary>
public class Customer : IEntity
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string Email { get; set; }
    public DateTime CreatedDate { get; set; }
    public bool IsActive { get; set; }

    public Customer(string name, string email)
    {
        Name = name ?? throw new ArgumentNullException(nameof(name));
        Email = email ?? throw new ArgumentNullException(nameof(email));
        CreatedDate = DateTime.Now;
        IsActive = true;
    }

    public override string ToString()
    {
        return $"Customer[{Id}]: {Name} ({Email}) - Active: {IsActive}";
    }
}

File: Java/Enterprise/Base/Customer.java
package Enterprise.Base;

/**
 * Example domain entity extending Layer Supertype
 */
public class Customer extends DomainObject {
    private String name;
    private String email;

    public Customer(String name, String email) {
        super();
        this.name = name;
        this.email = email;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
        markModified();
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
        markModified();
    }

    @Override
    public String toString() {
        return "Customer{id=" + getId() + ", name='" + name + "', email='" + email + "'}";
    }
}
